src/certificate_transfer_integration.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Helper class for trusting ca chains."""
+
+import os
+import subprocess
+from pathlib import Path
+from typing import Callable, Union
+
+from charms.certificate_transfer_interface.v0.certificate_transfer import (
+    CertificateAvailableEvent,
+    CertificateRemovedEvent,
+    CertificateTransferRequires,
+)
+from ops import CharmBase, Object
+
+from constants import CERTIFICATE_TRANSFER_NAME
+
+LOCAL_CA_CERTS_PATH = Path("/usr/local/share/ca-certificates")
+BUNDLE_PATH = "/etc/ssl/certs/ca-certificates.crt"
+
+
+class CertTransfer(Object):
+    def __init__(
+        self,
+        charm: CharmBase,
+        container_name: str,
+        callback_fn: Callable,
+        cert_transfer_relation_name: str = CERTIFICATE_TRANSFER_NAME,
+        bundle_name: str = "ca-certificates.crt",
+    ):
+        super().__init__(charm, cert_transfer_relation_name)
+        self.charm = charm
+        self.container = charm.unit.get_container(container_name)
+        self.cert_transfer = CertificateTransferRequires(charm, cert_transfer_relation_name)
+        self.callback_fn = callback_fn
+        self.bundle_name = bundle_name
+
+        self.framework.observe(
+            self.cert_transfer.on.certificate_available, self._on_certificate_event
+        )
+        self.framework.observe(
+            self.cert_transfer.on.certificate_removed, self._on_certificate_event
+        )
+
+    @property
+    def ca_bundle(self) -> str:
+        bundle = set()
+        for relation in self.charm.model.relations.get(self.cert_transfer.relationship_name, []):
+            for unit in set(relation.units).difference([self.charm.app, self.charm.unit]):
+                if ca := relation.data[unit].get("ca"):
```

Comment:
Just a **NONBLOCKING** question:

what should we do with the field `ca_chain` in the integration databag?

Reply:
That's a good question. Not sure what this is supposed to look like, but the `ca_chain` is always empty (https://github.com/canonical/self-signed-certificates-operator/blob/main/src/charm.py#L315, https://github.com/canonical/lego-base-k8s-operator/pull/144/files#diff-b87e0b9d60a904bf436a9c0c80fd05c448045726acaacd298468bff53e76b2e5R175). Should we prefix the `ca` with the `ca_chain`? I copied this logic from grafana, it seems to work for them, but maybe there is a bug? Not sure

Reply:
I think that you don't need to trust the ca_chain if you trust the ca (if you trust the root certificate, aka ca cert, you trust the whole chain).

I think that you need to trust the ca_chain only in cases where the ca cert is not in your trust store. I would suggest we leave it as it is and add it if needed.